xen/arch/x86/tpm.c

```diff
@@ -961,8 +961,12 @@ void tpm_hash_extend(unsigned loc, unsigned pcr, uint8_t *buf, unsigned size,
 }
 
 #ifdef __EARLY_TPM__
-void __stdcall tpm_extend_mbi(uint32_t *mbi)
+void __stdcall tpm_extend_mbi(uint32_t *mbi, uint32_t param)
 {
+    /* Early TPM code isn't linked with the rest but still needs to have this
+     * variable holding correct value. */
+    slaunch_param = param;
```

Comment:
I'll have to check if this links properly, early code is built differently.

Reply:
There is second definition of `slaunch_param` at the top inside `#ifdef __EARLY_TPM__` to take this into account.

Reply:
Ah, missed that, now it makes sense.